src/systems/a320_systems/src/pneumatic/wing_anti_ice.rs

```diff
@@ -0,0 +1,618 @@
+            wai_right_high_pressure_id: context
+                .get_identifier("PNEU_2_WING_ANTI_ICE_HIGH_PRESSURE".to_owned()),
+            wai_left_low_pressure_id: context
+                .get_identifier("PNEU_1_WING_ANTI_ICE_LOW_PRESSURE".to_owned()),
+            wai_right_low_pressure_id: context
+                .get_identifier("PNEU_2_WING_ANTI_ICE_LOW_PRESSURE".to_owned()),
+
+            wai_left_pressure_id: context
+                .get_identifier("PNEU_1_WING_ANTI_ICE_CONSUMER_PRESSURE".to_owned()),
+            wai_right_pressure_id: context
+                .get_identifier("PNEU_2_WING_ANTI_ICE_CONSUMER_PRESSURE".to_owned()),
+            wai_left_temperature_id: context
+                .get_identifier("PNEU_1_WING_ANTI_ICE_CONSUMER_TEMPERATURE".to_owned()),
+            wai_right_temperature_id: context
+                .get_identifier("PNEU_2_WING_ANTI_ICE_CONSUMER_TEMPERATURE".to_owned()),
+        }
+    }
+
+    fn is_wai_system_on(&self) -> bool {
+        self.wai_system_on
+    }
+
+    fn update_pressure_above_minimum(
+        &mut self,
+        context: &UpdateContext,
+        precooler_pressure: Pressure,
+        number: usize,
+    ) {
+        // The WAI valves open at min 10psi of pressure (difference)
+        self.wai_bleed_pressurised[number] =
+            precooler_pressure > context.ambient_pressure() + Pressure::new::<psi>(10.);
+    }
+
+    fn update_valve_controller(&mut self, context: &UpdateContext, number: usize) {
+        self.wai_valve_controller[number]
+            .valve_pid
+            .next_control_output(
+                self.wai_consumer_pressure(number).get::<psi>(),
+                Some(context.delta()),
+            );
+
+        self.wai_valve_controller[number].update(
+            context,
+            self.wai_relay.signals_on(),
+            self.wai_bleed_pressurised[number],
+        );
+    }
+
+    pub fn is_wai_valve_closed(&self, number: usize) -> bool {
+        !self.wai_valve[number].is_open()
+    }
+
+    pub fn is_precoooler_pressurised(&self, number: usize) -> bool {
+        self.wai_bleed_pressurised[number]
+    }
+
+    pub fn wai_consumer_pressure(&self, number: usize) -> Pressure {
+        self.wai_consumer[number].pressure()
+    }
+
+    pub fn wai_consumer_temperature(&self, number: usize) -> ThermodynamicTemperature {
+        self.wai_consumer[number].temperature()
+    }
+
+    pub fn wai_mass_flow(&self, number: usize) -> MassRate {
+        self.wai_exhaust[number].fluid_flow()
+    }
+
+    pub fn wai_timer(&self) -> Duration {
+        self.wai_relay.get_timer()
+    }
+
+    pub fn wai_valve_controller_on(&self, number: usize) -> bool {
+        self.wai_valve_controller[number].controller_signals_on()
+    }
+
+    // This is where the action happens
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        engine_systems: &mut [EngineBleedAirSystem; 2],
```

Comment:
Because `EngineBleedAirSystem` implements `PneumaticContainer` it's cleaner to pass `[&mut impl PneumaticContainer; 2]` instead. This also allows you to just pass it to the `update_move_fluid_with_transfer_speed` function.